src/components/ObsDetails/MasonryLayout.js

```diff
@@ -1,40 +1,66 @@
 import { ScrollView, View } from "components/styledComponents";
 import React, { useEffect, useState } from "react";
+import { ImageStyle, StyleProp } from "react-native";
 import Photo from "realmModels/Photo";
-import getImageDimensions from "sharedHelpers/getImageDimensions";
+import Sound from "realmModels/Sound";
+import type { RealmPhoto, RealmSound } from "realmModels/types";
+import getImageDimensions, { ImageDimensions } from "sharedHelpers/getImageDimensions";
 
 import PhotoContainer from "./PhotoContainer";
 import SoundContainer from "./SoundContainer";
 
+type PhotoItem = Photo & RealmPhoto;
+type PhotoItemWithDimensions = PhotoItem & ImageDimensions;
+type SoundItem = Sound & RealmSound;
+type Item = PhotoItem | SoundItem;
+type OriginalIndex = { originalIndex: number };
+type HydratedItem = ( SoundItem | PhotoItemWithDimensions ) & OriginalIndex;
+
 const numColumns = 2;
 const spacing = 6;
 
-const photoUrl = photo => Photo.displayLocalOrRemoteLargePhoto( photo );
+const photoUrl = ( photo: PhotoItem ) => {
+  const url = Photo.displayLocalOrRemoteLargePhoto( photo );
+  if ( !url ) {
+    throw new Error( "No photo URL found" );
+  }
+  return url;
+};
+
+// Photos have a `url` property, sounds have a `file_url` property
+const isSound = ( item: Item ): item is SoundItem => (
+  "file_url" in item
+);
+
+interface Props {
+  items: Item[];
+  onImagePress: ( index: number ) => void;
+}
 
-const MasonryLayout = ( { items, onImagePress } ) => {
-  const [columns, setColumns] = useState(
+const MasonryLayout = ( { items, onImagePress }: Props ) => {
+  const [columns, setColumns] = useState<HydratedItem[][]>(
     Array.from( { length: numColumns }, () => [] )
   );
 
   useEffect( () => {
     const distributeItems = async () => {
-      const newColumns = Array.from( { length: numColumns }, () => [] );
+      const newColumns: HydratedItem[][] = ( Array.from( { length: numColumns }, () => [] ) );
 
       const itemPromises = items.map( async item => {
         // If a sound, just return it
-        if ( item.file_url ) {
+        if ( isSound( item ) ) {
           return item;
         }
         const imageDimensions = await getImageDimensions( photoUrl( item ) );
-        return { ...item, ...imageDimensions };
```

Comment:
This old code was potentially problematic, which TypeScript caught. The spread operator does _not_ copy instance methods.

```
> class Cat {
|     age = 10;
|     meow() {}
| }
undefined

> cat = new Cat()
Cat { age: 10 }

> { ...cat, name: "Kevin" }
{ age: 10, name: 'Kevin' }

> { ...cat, name: "Kevin" }.meow()
Uncaught TypeError: {(intermediate value)(intermediate value)}.meow is not a function

> Object.assign(cat, { name: "Kevin" }).meow()
undefined
```


So Realm methods on `item` would not be copied here, which is presumably not what we want. To remedy this, I switched this to use the `Object.assign` function which should do what we want.